common/flatpak-remote.c

```diff
@@ -477,6 +480,54 @@ flatpak_remote_set_default_branch (FlatpakRemote *self,
   priv->local_default_branch_set = TRUE;
 }
 
+/**
+ * flatpak_remote_get_main_ref:
+ * @self: a #FlatpakRemote
+ *
+ * Returns the main ref of this remote, if set.
+ *
+ * Returns: (transfer full): the main ref, or %NULL
+ *
+ * Since: 1.1.1
+ */
+char *
+flatpak_remote_get_main_ref (FlatpakRemote *self)
+{
+  FlatpakRemotePrivate *priv = flatpak_remote_get_instance_private (self);
+
+  if (priv->local_main_ref_set)
+    return g_strdup (priv->local_main_ref);
+
+  if (priv->dir)
+    return flatpak_dir_get_remote_main_ref (priv->dir, priv->name);
+
+  return NULL;
+}
+
+/**
+ * flatpak_remote_set_main_ref:
+ * @self: a #FlatpakRemote
+ * @main_ref: The new main ref
+ *
+ * Sets the main ref of this remote.
+ *
+ * Note: This is a local modification of this object, you must commit changes
+ * using flatpak_installation_modify_remote() for the changes to take
+ * effect.
+ *
+ * Since: 1.1.1
+ */
+void
+flatpak_remote_set_main_ref (FlatpakRemote *self,
+                                   const char    *main_ref)
+{
```

Comment:
Hmm, I wonder if flatpak should maintain this by itself, when creating origin remotes ?

Reply:
Flatpak already does that. I just added the setters because everything else in FlatpakRemote had them. gnome-software doesn't actually need them.

Happy to remove them if you think it's better without :)